x/programs/wasmlanche/src/types.rs

```diff
@@ -25,6 +25,14 @@ impl From<Box<[u8]>> for ProgramId {
     }
 }
 
+#[cfg(feature = "test")]
+impl ProgramId {
+    #[must_use]
+    pub fn new(bytes: Box<[u8]>) -> Self {
+        Self(bytes)
+    }
+}
```

Comment:
This should be a `From<Box<[u8]>> for ProgramId`. Prefer `From` implementation over singe argument constructors.